# Python Basics Variable

# Code:1

### Declare two variables, 'x' and 'y', and assign them integer values. Swap the values of these variables without using any temporary variable.

In [1]:
x = 35
y = 46

In [2]:
# Perfroming XOR operation with x and y variables
x = x ^ y  
y = x ^ y
x = x ^ y

### the XOR operation performs the swap Operation 
#### x = 35 and y = 46
##### binary values of x and y are 100011 and 101110 = 001101 i.e now x = 13 in decimal 
##### binary values of x and y are 001101 and 101110 = 100011 i.e now y = 35 in decimal
##### binary values of x and y are 001101 and 100011 = 101110 i.e now x = 46 in decimal

In [11]:
print('x:',x,'y:',y) # print function prints the information in ( )

x: 46 y: 35


#### output

###### Now the value after swapping is x = 46

###### Now the value after swapping is y = 35

# Code:2

## Create a program that calculates the area of a rectangle. Take the length and width as input from the user and store them in variables. Calculate and display the area.

In [4]:
base = float(input("enter the base: "))        # Enter the Base value
height = float(input("enter the height: "))    # Enter the height value

enter the base:  35
enter the height:  46


In [12]:
area_traingle = 0.5 * base * height            # Area of a Triangle = (1/2)*b*h    i.e b = base   and h = height

In [7]:
print(f"Area of Triangle is:{area_traingle}")  # print function prints the information in ( )

Area of Triangle is:805.0


#### Output

##### Area of Triangle is: 805.0

# Code:3

### Write a python program that converts temperatures from Celsius to Fahrenheit. Take the temperature in Celsius as input, store as input, store it in a variable, convert it to Fahrenheit, and display the result.

In [13]:
celsius = int(input("Enter the Temperature in Celsius: "))   # Enter the temp in celsius
fahrenheit = celsius * (9/5) + 32                            # converting fahrenheit to celsius 

Enter the Temperature in Celsius:  36


### formula for converting celsius to fahrenheit is:
#### f = (c * (9/5) )+ 32

### Similarly for converting fahrenheit to celsius is:
#### c = (f - 32)* (9/5)

In [10]:
print("The Temperatur in fahrenhiet:",fahrenheit)             # print function prints the information in ( )

The Temperatur in fahrenhiet: 96.8


### Output

#### The Temperature after converting from celsius to fahrenheit is : 96.8

# String Based Question

## Program:1

### Write a python program that takes string as input and prints the length of the string.

In [27]:
string_1 = input("Enter a string: ")   # Enter a sting
length_string1 = len(string_1)         # len() function is used in find the length of the string

Enter a string:  My name is Mohammed Suleman, My Aim is to be a valuable Data Scientist


#### The below string contains of about 70 words including spaces, commas
#### My name is Mohammed Suleman, My Aim is to be a valuable Data Scientist

In [28]:
print(length_string1)                  # print function prints the information in ( )

70


### Output
#### The length of {My name is Mohammed Suleman, My Aim is to be a valuable Data Scientist} is: 70

## Program:2

### create a program that takes a sentence form user and counts the number of vowels (a, e, i, o, u) in the string.

In [72]:
string_vowel = input("Enter a sentence: ").lower()   # enter a sentence:  lower() function is used to fetch the Upper Case letters from the sentence
vowel_count = 0                                      # keeping the vowel count 0 
vowels = ['a','e','i','o','u']                       # vowels to count from a sentence: {a ,e, i, o, u}

Enter a sentence:  tsitneicS ataD


In [73]:
for char  in string_vowel:
    if char in vowels:
        vowel_count += 1

#### for char in string_vowels:
* for loop is used to check on each character in the string_vowels in lowercase
#### for char in vowels:
* checks the character from the input sentence whether contains a vowels = ['a','e','i','o','u']
* if vowel found in character the condition becomes 'True' and the code inside the if block will be executed
* if the condition is 'True' the "vowel_count += 1" increments the value of "string_vowel" variable by 1

In [74]:
print("Number of Vowels in the sentence: ",vowel_count) # print function prints the information in ( )

Number of Vowels in the sentence:  5


### Output
#### From the sentence: My name is Mohammed Suleman, My Aim is to be a valuable Data Scientist
#### Number of Vowels in the sentence:  24

## Program:3

### Given a string, reverse the order of the characters using slicing and print the revresed string.

In [58]:
string_0 = ("Data Scientist")         # sample string

In [70]:
reversed_string = string_0[::-1]      # slicing and scaling operation is performend

#### slicing is done by the [start index, end index, step index]

#### scaling is done by the [start index, end index, step index]
* example: for negative index [-1:0:-1] this takes a step by -1

In [71]:
print(reversed_string)                # print function prints the information in ( )

tsitneicS ataD 


### Output
#### The reversed_string for "Data Scientist" is "tsitneicS ataD"

## Program:4

### Write a program that takes a string as input and cheks of it is a Palindrome(reads the same forwards and backwards)

In [82]:
string_a = input("Enter a string")                 # Enter a string as a input
string_out = string_a.lower().replace(" ", "")     # string_a.lower().replace(" ","") stores the updated string in string_out

Enter a string racecar


#### string_a.lower().replace(" "," ")  
* .lower() is used to include Uppercase letters from the string or characters or sentence
* .replace is used to remove spaces from the string or characters or sentence

In [83]:
if string_out == string_out[::-1]:
    print("The given string is a Palindrome")
else:
    print("The given string is not a Palindrome")

The given string is a Palindrome


#### conditional statements is used to checks the conditions in both forward and reverse 
* if string_out is equals to string_out[::-1], then the print function prints the if statement
  * string_out[::-1] -- negative scaling is done to reverse the string from negative indexes by a step of -1
* if string_out is not equals to string_out[::-1], then the print function prints the else statement

### Output
####  example for a palindrome:
##### input("A man a plan a canal panama")
    * The given string is a Palindrome 

#### example for not a palindrome
##### input("few examples are better then one best")
    * The given string is not a Plaindrome

## Program:5

### Create a program that takes a string as input and removes all the spaces from it. print the modified string without spaces.

In [108]:
string_b = input("Enter a string")
string_c = string_b.replace(" ","")                    

Enter a string the apple is red an it keeps doctor a w ay


#### string_b.replace() funtion is used to replce the parameters from one form to another
     * in this case we have replce the parameters from space to no space
            * replace(" ","") --- we removed the space from " " with no space "",  this removes the space from the string
            * We can use other parametrs to add any other character, white spaces, special character with the .replace() function


In [111]:
print(f"updated string_b-{string_b},is: {string_c}")

updated string_b-the apple is red an it keeps doctor a w ay,is: theappleisredanitkeepsdoctoraway


### Output
#### input(" the apple is red      an it keeps      doctor a w   ay")
      * 'theappleisredanitkeepsdoctoraway'